dcap/types/src/quote3.rs

```diff
@@ -378,6 +378,67 @@ impl<'a> TryFrom<&'a [u8]> for CertificationData<'a> {
     }
 }
 
+impl<'a> IntoIterator for &'a CertificationData<'a> {
+    type Item = &'a [u8];
+    type IntoIter = PemIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        PemIterator {
+            pem_data: self.data,
+        }
+    }
+}
+
+const BEGIN_PEM: &[u8] = b"-----BEGIN ";
+const END_PEM: &[u8] = b"-----END ";
+
+struct PemIterator<'a> {
+    pem_data: &'a [u8],
+}
+
+/// Iterator over each PEM in a provided buffer.
+impl<'a> Iterator for PemIterator<'a> {
+    type Item = &'a [u8];
+    fn next(&mut self) -> Option<&'a [u8]> {
+        if self.pem_data.is_empty() {
```

Comment:
Is there an expected, normal case where a `CertificationData` will hold no PEM certs at all? It seems like maybe this should be an error rather than silent failure but I don't fully understand how we're using this so it might make sense to just skip over it if it's empty here.

Reply:
I think for the _generic_ `PemIterator` it makes sense. (though I may be guilty of genericising too soon)

It may be that instead of the `IntoIterator` trait for `CertificationData` it is refactored to a fallible method of `into_pem_iter()` or similar.  
Though this may still be lacking full assurances.  At some point we will need to convert `PEM -> DER` and then verify the resultant cert chain.  I think the verification of the cert chain will be the most logical place to fail, no cert chain should be an "unverifiable" error.